utils/train_utils.py

```diff
@@ -126,7 +125,7 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
         if train_config.run_validation:
             eval_ppl, eval_epoch_loss = evaluation(model, train_config, eval_dataloader, rank, tokenizer)   
             if train_config.save_model and eval_epoch_loss < best_val_loss:
-                
+                dist.barrier()
                 if  train_config.use_peft:
                     
                     print(f"we are in the saving the PEFT modules")
```

Comment:
For the PEFT model saving, does this need to happen on all the ranks or only the rank 0? Please check

Reply:
as the model is wrapped in fsdp, this would not work, it hangs then after saving on rank0.